### DISCUSSION TITLE
Automatically resizing adjacent windows

### DISCUSSION THREAD

**Discussion by rxhanson** [rxhanson/RectanglePro-Community#429] [Show and tell]
When you resize a window by dragging the window edge, you can have Rectangle Pro automatically resize the window next to it. Check this box:
![image](https://github.com/rxhanson/RectanglePro-Community/assets/13651296/4c87c1eb-2cda-41af-bd58-7262468bf840)

Notice that the adjacent window will resize upon mouse up, not continuously as the window is resized. There is a limitation with the macOS accessibility API that makes resizing from the left edge wiggle around, and doing it on mouse up avoids this.

Additionally, if you use the resize on the window throw tab, you can check the box there to have the adjacent resize occur after resizing with modifier + cursor movement here.
![image](https://github.com/rxhanson/RectanglePro-Community/assets/13651296/63ca950e-bd6a-4446-946f-152ddd303a52)

**Comment by unlocomqx** [comment]
It would be cool if this also worked with the _Larger_ / _Smaller_ shortcuts. I saw something like this [here](https://x.com/wesbos/status/1857474924262146347)


https://github.com/user-attachments/assets/eed449b7-94e1-4bb8-877b-2f02169f3897

**Reply by rxhanson** [reply]
Oh nice, thanks. I'll see what I can do in a future release.

**Reply by withakay** [reply]
I was just pondering this, I have set key commands for larger and smaller width (via defaults) and was hoping the adjacent window would resize with that.

**Reply by antony4859** [reply]
Found this thread, any update on this?
Would be amazing if it did work!

**Reply by borekb** [reply]
I'll add here that I often use "Left Half" / "Righ Half" to cycle between various sizes (1/3, 1/2 and 2/3) and it would be great if adjacent window reacted to this as well.

**Reply by rxhanson** [reply]
For smaller/larger commands, there is this Terminal command to enable it, but it has a bug where holding the shortcut will lose the adjacent windows, hence it not being added into the UI yet:
```
defaults write com.knollsoft.Hookshot actionAdjacentResizing -int 1
```
Restart the app after executing that.

@borekb I have this backlog item and will add your +1 to it, thanks!

**Comment by dseehof** [comment]
Maybe I'm doing something wrong, but on the latest Sequoia 15.2 and Rectange Pro 3.0.37 I don't get any adjacent windows to resize at all.

> This is working for me. Perhaps try restarting your Mac one more time since the update if you haven't done so already. You could also try quitting the app, removing the accessibility permissions, then relaunching the app and re-adding them as prompted.

To be honest it didn’t work for a number of weeks now. I tried all your steps to no avail – funny, as everything else I use works just fine.

**Reply by rxhanson** [reply]
Would you mind posting the logs for when this happens? You can show logs by holding the option key in the menu bar menu and selecting View Logging.

**Comment by dseehof** [comment]
Sure thing. Two terminal windows side by side, this is what I get.

2024-12-10T08:49:10+01:00: Updating visible window info
2024-12-10T08:49:11+01:00: Adjusting adjacent to window with frame: (1841.0, -1667.0, 1755.0, 1649.0)
2024-12-10T08:49:11+01:00: resized edges: [Rectangle_Pro.EdgeDelta(edge: Rectangle_Pro.RectEdge.minX, initial: 2091.0, final: 1841.0, delta: -250.0, segment: Rectangle_Pro.RectSegment(orientation: Rectangle_Pro.SegmentOrientation.vertical, edge: Rectangle_Pro.RectEdge.minX, extrema: Rectangle_Pro.Extrema.min, zIndex: nil, windowInfo: nil, min: -1667.0, max: -18.0, k: 2091.0))]
2024-12-10T08:49:11+01:00: affected windows: []
2024-12-10T08:49:11+01:00: additional windows: []
2024-12-10T08:49:11+01:00: adjusted windows: [:]

**Reply by rxhanson** [reply]
Interesting, thanks! This might still be due to accessibility privileges acting strange. Just so we can rule that out, let's try a more in-depth cleaning out of all things Rectangle Pro, if you don't mind.

1. Quit Rectangle Pro
2. Execute this Terminal command: 
tccutil reset All com.knollsoft.Hookshot
3. Delete Rectangle Pro
4. Reboot
5. Reinstall Rectangle Pro
6. Launch Rectangle Pro and give it accessibility permissions again.

**Reply by chodorowicz** [reply]
This feature works for me only on my main screen, but not on the external ones.

**Reply by rxhanson** [reply]
@chodorowicz What is your external monitor setup? Can you send a screenshot of System Settings > Displays > Arrange...? Is there anything unusual about your displays?

**Reply by chodorowicz** [reply]
Sure!
BenQ PD3200U - 3200 × 1800
Iiyama PL2893UH - 2560 × 1440

I've tried both in a different setup with an external Arzopa screen (which works fine), and windows on those former were not being adjusted when paired in a different setup too.

I'm using BetterDisplay but no change after quitting it. 

![CleanShot 2025-01-07 at 10  37 25@2x](https://github.com/user-attachments/assets/d00bf6a8-e465-454f-b178-2678c681a2a4)

**Comment by dseehof** [comment]
Did all that, in the order you suggested. It did ask for permissions again which I granted. But no change, log looks identical too. 🤷‍♂️
